From ad5d2dfec2443e74a707615fcf97bbd17fd4bde5 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 01 Apr 2022 14:24:00 +0300
Subject: [PATCH] [SW-77251] gaudi3: check die id in erom function

EROM code is expected to be executed by master die, this
patch add check of die id in EROM recovery function.

Change-Id: I25905babd7a9b93d87118cf2aede09c89ee3d774
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 9161456..c094a00 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -40,6 +40,10 @@
 
 static void execute_erom_recovery(void)
 {
+	/* EROM recovery not supported in slave die */
+	if (arcboot_rom_global.die_id)
+		return;
+
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_EROM_RECOVERY);
 	/* TODO: implement erom recovery */
 }
